support.ts

```diff
@@ -1,45 +1,56 @@
 // @ts-ignore
-import { sparqlEscapeUri, sparqlEscapeString } from "mu";
-import fetch from "node-fetch";
-import { LDES_ENDPOINT, LDES_FRAGMENTER } from "./config";
+import { sparqlEscapeUri, sparqlEscape } from "mu";
+import { LDES_FOLDER, LDES_FRAGMENTER, } from "./config";
 import { Changeset, Quad, Term } from "./types";
+import { addData, getConfigFromEnv } from "@lblod/ldes-producer";
 
+const ldesProducerConfig = getConfigFromEnv();
+console.log("ldesProducerConfig", ldesProducerConfig);
 
-export function toSparqlTerm(thing: Term): string {
-	if (thing.type == "uri") return sparqlEscapeUri(thing.value);
-	else return sparqlEscapeString(thing.value);
-}
-
-export function toSparqlTriple(quad: Quad): string {
-	return `${toSparqlTerm(quad.subject)} ${toSparqlTerm(
-		quad.predicate
-	)} ${toSparqlTerm(quad.object)}.`;
-}
+const datatypeNames = {
+	"http://www.w3.org/2001/XMLSchema#dateTime": "dateTime",
+	"http://www.w3.org/2001/XMLSchema#date": "date",
+	"http://www.w3.org/2001/XMLSchema#decimal": "decimal",
+	"http://www.w3.org/2001/XMLSchema#integer": "int",
+	"http://www.w3.org/2001/XMLSchema#float": "float",
+	"http://www.w3.org/2001/XMLSchema#boolean": "bool",
+};
+const sparqlEscapeObject = (bindingObject: Term): string => {
+	const escapeType = datatypeNames[bindingObject?.datatype || ""] || "string";
+	return bindingObject.type === "uri"
+		? sparqlEscapeUri(bindingObject.value)
+		: sparqlEscape(bindingObject.value, escapeType);
+};
 
-async function sendLDESRequest(uri: string, body: string) {
-	const queryParams = new URLSearchParams({
-		resource: uri,
-		...(LDES_FRAGMENTER && { fragmenter: LDES_FRAGMENTER })
-	});
 
-	return fetch(`${LDES_ENDPOINT}?` + queryParams, {
-		method: "POST",
-		headers: {
-			"Content-Type": "text/turtle",
-		},
-		body: body,
-	});
+export function toSparqlTriple(quad: Quad): string {
+	return `${sparqlEscapeObject(quad.subject)} ${sparqlEscapeObject(quad.predicate)} ${sparqlEscapeObject(quad.object)}.`;
 }
 
-export async function moveTriples(changesets: Changeset[]) {
+export async function moveTriples(changesets: Changeset[], stream = LDES_FOLDER) {
+	let turtleBody = "";
```

Comment:
now that you add the sparqlEscape things, they will write ^^xsd strings so you need the prefix because else the library complains about the prefix being undefined.
```suggestion
	let turtleBody = "@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .\n";
```

Reply:
added the prefix